src/FtlServer.cpp

```diff
@@ -166,8 +166,11 @@ Result<uint16_t> FtlServer::onControlStartMediaPort(FtlControlConnection& contro
     {
         throw std::runtime_error("Unknown control connection requested a media port assignment");
     }
-    std::unique_ptr<FtlControlConnection> control = std::move(pendingControlConnections[&controlConnection]);
-    pendingControlConnections.erase(&controlConnection);
+
+    // Don't erase the connection from the pending store just yet -
+    // if there's an error assigning it a port, we need to keep it around so it can handle it!
+    std::unique_ptr<FtlControlConnection>& control = 
+        pendingControlConnections.at(&controlConnection);
```

Comment:
Won't this cause an issue on line 200?

Honestly would expect the compiler to catch the issue, looks like creating the FtlStream consumes the control connection unique_ptr while the ptr is also still in the map. My C++ is rusty but that seems bad, maybe the `std::move(control)` on the lvalue reference bypasses the compiler safety?

Reply:
This is a reference to the unique_ptr inside of the map, so no construction is happening afaik. At line 200, std::move causes the move constructor for unique_ptr to be used to transfer ownership of the the unique_ptr to the new FtlStream, so when `control` leaves scope, it's empty.

Reply:
Ahhhhh yeah that makes sense. An empty unique_ptr basically acts as a nullptr right so this is much more normal than I thought, thanks!